modules/ROOT/pages/subqueries/subqueries-in-transactions.adoc

```diff
@@ -550,3 +550,83 @@ RETURN largeLists
 1+d|Rows: 2
 
 |===
+
+[[composite-databases]]
+== Composite databases
+`CALL { ... } IN TRANSACTIONS` works on composite databases.
+Composite databases are virtual databases used for querying their constituents, so the examples above are generally
+valid for composite databases, but they must be slightly modified to accommodate the nature of composite databases.
+
+The following two examples shows how `CALL { ... } IN TRANSACTIONS` can be used on a composite database.
+Both examples are the equivalent of examples presented in the previous sections, but modified for composite databases.
+The first query uses 'friends.csv' from previous examples and creates the person nodes on all constituents.
+
+.friends.csv
+[source, csv, role="noheader" filename="friends.csv"]
+----
+1,Bill,26
+2,Max,27
+3,Anna,22
+4,Gladys,29
+5,Summer,24
+----
+
+.Query
+[source, cypher]
+----
+UNWIND graph.names() AS graphName
+LOAD CSV FROM 'file:///friends.csv' AS line
+CALL {
+  USE graph.byName( graphName )
+  WITH line
+  CREATE (:Person {name: line[1], age: toInteger(line[2])})
+} IN TRANSACTIONS
+----
+
+The second query removes all nodes from all constituents.
+
+.Query
+[source, cypher]
+----
+UNWIND graph.names() AS graphName
+CALL {
+  USE graph.byName( graphName )
+  MATCH (n)
+  RETURN id(n) AS id
+}
+CALL {
+  USE graph.byName( graphName )
+  WITH id
+  MATCH (n)
+  WHERE id(n) = id
+  DETACH DELETE n
+} IN TRANSACTIONS
+----
+
+The two examples demonstrate another useful feature of `CALL { ... } IN TRANSACTIONS`.
+Even though composite databases allow accessing multiple graphs in a single query,
+only one graph can be modified in one transaction. `CALL { ... } IN TRANSACTIONS`
+offers a way how to construct queries modifying multiple graphs.
+
+[[composite-databases-batching]]
+=== Batch size in composite databases
+Since queries on composite databases can interact with more than one graph, batching is slightly more complicated there.
```

Comment:
```suggestion
Since queries on composite databases can interact with more than one graph, batching behaves differently.
```

Reply:
I don't like the word "differently". The declared batch size is an upper limit on non-composite databases, too. If you declare batch size `x`, it does not mean that it is always going to be `x`. For example, if you have batch size of 2 and there are 5 input rows, the last batch will be 1.

Reply:
ok, how about "complex" instead of "complicated"?

Reply:
done